### ISSUE TITLE
Show commit count based on revision

### ISSUE THREAD

**Issue by MeBrei** [radicle-dev/radicle-upstream#924]
The commit tab currently shows the number of commits from the project stats. Instead the number of commits on the selected revision should be shown.

**Comment by MeBrei** [comment]
A quick solution for this would be to just count the number of commits returned by the `commits` endpoint in the ui for the given revision. Should we implement pagination for this endpoint at some point, it should return the total number of items along with the batch of commits.

From what I understand, the number of commits could also be calculated in the proxy, by using `git::Browser` for the specific branch and then getting the statistics via `browser.get_stats`, but that also just calls the length via `history.len()`.

For now I would go for the quick solution.

**Comment by cloudhead** [comment]
Sounds good to me.

**Comment by FintanH** [comment]
To clarify, is it the case that if we switch revisions then the wrong commit count is shown? If so, that sounds like a bug in `radicle-surf`

**Comment by MeBrei** [comment]
@FintanH  The commit count in the ui is taken from `project.stats.commits`. This is returned by the `project` endpoint and only called when first loading the `Project` page. The stats in proxy are always created with the default branch (ie the `Browser`)

**Comment by rudolfs** [comment]
@FintanH to provide a bit more context: the UI got re-shuffled, so that the revision selector is visually on the same level as the "Source" and "Commits" tabs -- so intuitively we (@juliendonck, @brandonhaslegs and I) were thinking that the number should be tied to the currently selected revision. And clicking on the "Commits" tab leads us to a list of the commits in the currently selected revision.

<img width="797" alt="Screenshot 2020-09-22 at 11 42 46" src="https://user-images.githubusercontent.com/158411/93867225-06be9200-fcc9-11ea-895a-898e92c7527a.png">

**Comment by FintanH** [comment]
> @FintanH The commit count in the ui is taken from project.stats.commits. This is returned by the project endpoint and only called when first loading the Project page. The stats in proxy are always created with the default branch (ie the Browser)

Ah, I see.

> were thinking that the number should be tied to the currently selected revision. And clicking on the "Commits" tab leads us to a list of the commits in the currently selected revision.

That's how I'd expect it to work as well :sweat_smile:

My personal preference for solving this would be introducing an endpoint in `source` such as `stats/<project urn>/revision` and it creates the `Browser` based off of the urn and the revision. It should be a straight-forward addition.

**Comment by rudolfs** [comment]
Another option would be to add the counts to the `revisions` endpoint:
<details>
<summary>curl -X GET "http://localhost:8080/v1/source/revisions/rad:git:hwd1yredhfxaf1kdynsyas4zxfzayx5oohj3w6wcp5yd6gx4gebb5nfghey"
</summary>

```json
[
  {
    "identity": {
      "peerId": "hynqf6ptbyafrd1trrf713duz8zpksgd1xfj8oj8wxbp9jecbajjzh",
      "urn": "rad:git:hwd1yrenyjj1nq8rrtwebdrg49gmkob4cbk8uwcnd859435n3g416x5of8a",
      "shareableEntityIdentifier": "ruudis@hynqf6ptbyafrd1trrf713duz8zpksgd1xfj8oj8wxbp9jecbajjzh",
      "metadata": { "handle": "ruudis" },
      "avatarFallback": {
        "emoji": "😃",
        "background": { "r": 152, "g": 77, "b": 170 }
      }
    },
    "branches": ["master"],
    "tags": []
  }
]
```
</details>

Something like:
```json
...
    "branches": [
      {"name": "master", "commitCount": 345},
      {"name": "dev", "commitCount": 12}
    ],
    "tags": []
...
```

**Comment by MeBrei** [comment]
I would also prefer the stats/commitCount to be included in an existing request, either `revisions` or `commits`. 
I would have to investigate though, if getting the commit count adds a significant time to either of the requests. (ie the Header of the project page waits for the `revisions` data)

**Comment by rudolfs** [comment]
> if getting the commit count adds a significant time to either of the requests

Yeah, I'm wondering about that too. If we take the git implementation (dunno how surf/libgit2 implements this) for reference, the time could be substantial for bigger repos:

```sh
work/radicle-upstream » time git rev-list --count master
491
git rev-list --count master  0.01s user 0.01s system 65% cpu 0.026 total

work/radicle-upstream » time git rev-list --count rudolfs/ui-fixes
498
git rev-list --count rudolfs/ui-fixes  0.01s user 0.01s system 81% cpu 0.017 total

work/linux » time git rev-list --count master
950540
git rev-list --count master  6.53s user 0.30s system 99% cpu 6.830 total
```

**Comment by MeBrei** [comment]
In #962 I decided to add the stats into the commit endpoint. This way I could connect waiting for the commits to load and showing the commit count. This could also be used in #920, ie disabling the tab (or a different loading state) while waiting for the data.
